agent/command/test_selection_get.go

```diff
@@ -150,11 +151,14 @@ func (c *testSelectionGet) Execute(ctx context.Context, comm client.Communicator
 	startTime := time.Now()
 	selectedTests, err := comm.SelectTests(ctx, conf.TaskData(), request)
 	durationMs := time.Since(startTime).Milliseconds()
+	calledAPI = true
 	trace.SpanFromContext(ctx).SetAttributes(attribute.Int64(testSelectionDurationMsAttribute, durationMs))
-	trace.SpanFromContext(ctx).SetAttributes(attribute.Bool(testSelectionCalledAttribute, true))
 	if err != nil {
 		return errors.Wrap(err, "calling test selection API")
 	}
+	defer func() {
```

Comment:
This defer has to be called higher up, otherwise it won't be set at all in cases like if `!enabled` or if the TSS call errors.

Reply:
I moved it up, and verified that the closure does indeed update the calledAPI var 

https://ui.honeycomb.io/mongodb-4b/environments/staging/datasets/evergreen-agent/result/pQDayJzgthm/trace/BzeYqWEYtiu?fields%5B%5D=s_name&fields%5B%5D=s_serviceName&span=9d3c0d4c0b0933f8